src/script/interpreter.cpp

```diff
@@ -60,6 +61,67 @@ static inline void popstack(std::vector<valtype>& stack)
     stack.pop_back();
 }
 
+static inline int64_t cast_signed64(uint64_t v)
+{
+    uint64_t int64_min = (uint64_t)std::numeric_limits<int64_t>::min();
+    if (v >= int64_min)
+        return static_cast<int64_t>(v - int64_min) + int64_min;
+    return static_cast<int64_t>(v);
+}
+
+static inline int64_t read_le8_signed(const unsigned char* ptr)
+{
+    return cast_signed64(ReadLE64(ptr));
+}
+
+static inline void push4_le(std::vector<valtype>& stack, uint32_t v)
+{
+    valtype vch;
+    auto v_le = htole32(v);
+    vch.insert(vch.begin(), (unsigned char*)&v_le, (unsigned char*)&v_le + 4);
+    stack.push_back(vch);
+}
+
+static inline void push8_le(std::vector<valtype>& stack, int64_t v)
```

Comment:
Probably should be `uint64_t v` parameter.

Reply:
Sure, there will be implicit cast from `signed` to `unsigned` somewhere. Don't really have a preference whether it's in a function body or parameter.